lib/validate.js

```diff
@@ -4,16 +4,15 @@ const {Table} = require('tableschema')
 
 const {File, isUrl} = require('data.js')
 
-async function validate(descriptor, basePath) {
+async function validate(datasetObj) {
   try {
-    await validateMetadata(descriptor)
-    for (let i = 0; i < descriptor.resources.length; i++) {
-      // TODO what if resource is remote
-      const resource = File.load(descriptor.resources[i], {basePath})
-      if (resource.descriptor.format === 'csv') {
+    await validateMetadata(datasetObj.descriptor)
+    for (let i = 0; i < datasetObj.resources.length; i++) {
+      if (datasetObj.resources[i].descriptor.format === 'csv') {
         await validateData(
-          resource.descriptor.schema,
-          resource.path
+          datasetObj.resources[i].descriptor.schema,
+          datasetObj.resources[i].path,
+          datasetObj.resources[i].descriptor.dialect
```

Comment:
What about other stuff like `escapeChar`, `seperator`, or that is included in the schema?

Reply:
Also, can we handle skipping rows? Will a user be able to push file that needs to skip rows?

Reply:
@zelima all parser options are included in the `datasetObj.resources[i].descriptor.dialect`

Re skipping rows, when validating data we use tableschema library and I don't see any option there to skip rows.